### PR TITLE
[dicp][tops] Adjust for llama_finetune.

### PR DESCRIPTION
**Pull request by Reinerzhou**

### CODE REVIEW

File: dicp/dicp/vendor/TopsGraph/codegen/enflame.py [hunk 1]
```diff
@@ -381,20 +383,19 @@ def check_res(a, b, graph_name):
...
-            """, strip=True
-        )
+            """,
+                                  strip=True)
         return compile_graph_code.getvalue()
 
     def gen_tensor(self, prefix, tensor):
         if dipu_flag:
-            res = f"{prefix}({tuple(tensor.shape)}, {tensor.stride()}, device='dipu:{self.device_id}', dtype={tensor.dtype})"
+            res = f"{prefix}({tuple(tensor.shape)}, {tensor.stride()}, device='xpu:{self.device_id}', dtype={tensor.dtype})"
```
Comment (jinminxi104) [inline dicp/dicp/vendor/TopsGraph/codegen/enflame.py:398]: 统一dipu

**Review by jinminxi104 — CHANGES_REQUESTED** [review]

File: dicp/dicp/vendor/TopsGraph/conversion.py [hunk 2]
```diff
@@ -210,14 +210,10 @@ def HardswishBackward(self, *args, **kwargs):
     def Clone(self, *args, **kwargs):
         return self.get_proxy(tops_op.Clone, args, kwargs)
 
-    @register_conversion(aten.copy.default)
+    @register_conversion([aten.copy.default, aten.copy_.default])
```
Comment (CyCle1024) [inline dicp/dicp/vendor/TopsGraph/conversion.py:213]: add comment that warns for situation besides inplace copyting the input args.

**Review by CyCle1024 — CHANGES_REQUESTED** [review]

**Review by jinminxi104 — APPROVED** [review]

File: dicp/dicp/vendor/TopsGraph/codegen/enflame.py [hunk 3]
```diff
@@ -902,8 +901,8 @@ def Concatenate(op_var, out_shape, out_dtype, tensors, dim):
         return f"builder::Op {op_var} = builder::Concatenate({'{' + ', '.join(tensors) + '}'}, {dim});"
 
     @staticmethod
-    def Softmax(op_var, out_shape, out_dtype, x, y, z):
-        return f"builder::Op {op_var} = builder::Softmax({x}, {y}, {z});"
+    def Softmax(op_var, out_shape, out_dtype, x, y):
+        return f"builder::Op {op_var} = builder::Softmax({x}, {y}, true);"
```
[the remark this answers is not in the document]
Comment (Reinerzhou) [reply dicp/dicp/vendor/TopsGraph/codegen/enflame.py:906]: done.

File: dicp/dicp/vendor/TopsGraph/conversion.py (hunk 2 above)
Comment (Reinerzhou) [reply dicp/dicp/vendor/TopsGraph/conversion.py:214]: done.

**Review by CyCle1024 — APPROVED** [review]

**Pull request merged**